File: apps/mobile-flutter/test/unit/mock_data_test.dart
import 'package:careconnect_mobile/core/utils/date_formatting.dart';
import 'package:careconnect_mobile/data/mock_data.dart';
import 'package:careconnect_mobile/models/dose_event.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  final day = DateTime(2026, 8, 25, 14, 14);
  final seed = MockData.seed(day);

  test('seeds relative to the given day', () {
    expect(seed.seededOn, DateTime(2026, 8, 25));
    final today = seed.doseEvents
        .where((d) => DateFormatting.isSameDay(d.scheduledFor, day))
        .toList();
    expect(today, hasLength(5));
    final yesterday = seed.doseEvents
        .where(
          (d) =>
              DateFormatting.isSameDay(d.scheduledFor, DateTime(2026, 8, 24)),
        )
        .toList();
    expect(yesterday, hasLength(5));
    expect(yesterday.every((d) => d.status == DoseStatus.taken), isTrue);
  });

  test('matches the Week 3 Figma story at 2:14 PM', () {
    final metformin = seed.doseById('dose-metformin')!;
    expect(metformin.scheduledFor, DateTime(2026, 8, 25, 14, 34));
    expect(metformin.isDue, isTrue);
    final atorvastatin = seed.doseById('dose-atorvastatin')!;
    expect(atorvastatin.isOverdue(day), isTrue);
    expect(seed.doseById('dose-lisinopril-am')!.status, DoseStatus.taken);
    expect(seed.doseById('dose-vitamin-d')!.status, DoseStatus.skipped);
    expect(
      seed.appointments.map((a) => a.title),
      contains('Dr. Alvarez — Cardiology follow-up'),
    );
  });

  test('every schedule time parses', () {
    for (final med in seed.medications) {
      for (final time in med.scheduleTimes) {
        expect(() => DateFormatting.parseLocalTime(time), returnsNormally);
      }
      expect(med.patientId, MockData.patientId);
    }
  });

  test('ids are unique across every collection', () {
    final ids = [
      ...seed.medications.map((m) => m.id),
      ...seed.doseEvents.map((d) => d.id),
      ...seed.appointments.map((a) => a.id),
      ...seed.activity.map((a) => a.id),
    ];
    expect(ids.toSet().length, ids.length);
  });

  test('data is obviously fictional — no real PHI', () {
    expect(seed.patient.email, endsWith('@example.test'));
    expect(seed.caregiver.phone, startsWith('555-01'));
    expect(seed.caregiver.patientIds, [seed.patient.id]);
    expect(seed.patient.primaryCaregiverId, seed.caregiver.id);
  });

  test('is a pure function of the day', () {
    final again = MockData.seed(DateTime(2026, 8, 25, 3));
    expect(again.doseEvents, equals(seed.doseEvents));
    final other = MockData.seed(DateTime(2026, 9, 6));
    expect(
      other.doseById('dose-metformin')!.scheduledFor,
      DateTime(2026, 9, 6, 14, 34),
    );
  });
}

File: apps/react-mobile/src/data/mockData.test.ts
import { isSameDay } from '../core/utils/dateFormatting';
import { activeMedications, medicationById } from '../models/domain';
import { CAREGIVER_ID, PATIENT_ID, seedCareData } from './mockData';

describe('seedCareData', () => {
  const seedDay = new Date(2026, 7, 25, 14, 14);
  const data = seedCareData(seedDay);

  it('seeds the patient and caregiver from the mock roster', () => {
    expect(data.patient.id).toBe(PATIENT_ID);
    expect(data.caregiver.id).toBe(CAREGIVER_ID);
    expect(data.patient.primaryCaregiverId).toBe(CAREGIVER_ID);
  });

  it('seeds four active medications', () => {
    expect(activeMedications(data)).toHaveLength(4);
  });

  it('seeds doses for both yesterday and today', () => {
    const yesterday = new Date(2026, 7, 24);
    const hasYesterday = data.doseEvents.some((d) => isSameDay(d.scheduledFor, yesterday));
    const hasToday = data.doseEvents.some((d) => isSameDay(d.scheduledFor, seedDay));
    expect(hasYesterday).toBe(true);
    expect(hasToday).toBe(true);
  });

  it('matches the Week 3 Figma story: Metformin due soon, Atorvastatin overdue', () => {
    const metformin = data.doseEvents.find((d) => d.id === 'dose-metformin')!;
    const atorvastatin = data.doseEvents.find((d) => d.id === 'dose-atorvastatin')!;
    expect(metformin.status).toBe('due');
    expect(metformin.scheduledFor.getTime()).toBeGreaterThan(seedDay.getTime());
    expect(atorvastatin.status).toBe('due');
    expect(atorvastatin.scheduledFor.getTime()).toBeLessThan(seedDay.getTime());
  });

  it('seeds three upcoming appointments', () => {
    expect(data.appointments).toHaveLength(3);
    expect(medicationById(data, 'med-metformin')?.name).toBe('Metformin');
  });

  it('seeds a non-empty activity feed', () => {
    expect(data.activity.length).toBeGreaterThan(0);
  });

  it('is deterministic for the same seed day', () => {
    const again = seedCareData(seedDay);
    expect(again.doseEvents).toEqual(data.doseEvents);
  });
});
